lua/otter/keeper.lua

```diff
@@ -350,9 +351,7 @@ M.sync_raft = function(main_nr, lang)
             end
           end
 
-          -- clear buffer
-          api.nvim_buf_set_lines(otter_nr, 0, -1, false, {})
```

Comment:
This is still necessary, though I also would like to find a more elegant approach, because otherwise code cells deleted at the end of the main buffer never get deleted in the otter buffer, so you end up with their diagnostics also never being deleted:
![image](https://github.com/jmbuhr/otter.nvim/assets/17450586/e54f74b8-da13-4d72-b376-2eef0b410bbd)

![image](https://github.com/jmbuhr/otter.nvim/assets/17450586/387cde1e-d481-47b2-b7da-2bfc10efa043)

Reply:
Ooh I see. Pretty sure this has an easy fix. Will take a look later today after I wake up

Reply:
maybe just clear from `nmax` to `-1`. This should not lead to the flickering you found.

Reply:
I was thinking change nmax to -1. does that work?

Reply:
Hmmm. I'm seeing that sync raft just isn't getting to the line of code in question when deleting chunks. Might have a different bug

Reply:
This is the other bug. When you remove all the code chunks, this happens before we clear the otter buffer. So we still need to clear the otter buffer in this case.
 
```lua
if next(all_code_chunks) == nil then
  return
end
```